trio/_channel.py

```diff
@@ -306,15 +342,15 @@ async def receive(self):
         self._state.receive_tasks[task] = None
         task.custom_sleep_data = self
 
-        def abort_fn(_):
+        def abort_fn(_) -> Abort:
             self._tasks.remove(task)
             del self._state.receive_tasks[task]
             return trio.lowlevel.Abort.SUCCEEDED
 
-        return await trio.lowlevel.wait_task_rescheduled(abort_fn)
+        return await trio.lowlevel.wait_task_rescheduled(abort_fn)  # type: ignore
```

Comment:
Why does this need a `# type: ignore` but not the other similar pattern in `MemorySendChannel`.

Reply:
`MemorySendChannel.send()` doesn't return anything, which this method does. The problem is that `wait_task_rescheduled` isn't typed so we get an `Any` as return type, which breaks us having specified the method to return `ReceiveType`.

When I looked into `trio-typing` it explicitly says that `wait_task_rescheduled` returns `Any`
```python
async def wait_task_rescheduled(
    abort_func: Callable[[Callable[[], NoReturn]], Abort]
) -> Any: ...
```
https://github.com/python-trio/trio-typing/blob/d5835f7065f58ebdd95ca27b09d1d5cbe686b8b8/trio-stubs/lowlevel.pyi#L131-L133

but I didn't manage to figure out whether that's a limitation in the typing, or what's up with that. (as I admitted in my first post).

Reply:
We could probably just move the `return` to a new line. Let's verify with `wait_task_rescheduled` first though, maybe? Perhaps this is old code.

Reply:
Moved it to a new line, but yeah `wait_task_rescheduled` need to be figured out.